### DISCUSSION TITLE
How to enable easy installation of an ipywidget as jupyter extension

### DISCUSSION THREAD

**Discussion by paddymul** [microsoft/vscode-discussions#2362] [Extension Development QnA]
My project - [Buckaroo](https://github.com/paddymul/buckaroo) works in VSCode notebooks last I checked.  I'm trying to figure out what is necessary to make it installable as a VSCode from the VSCode extension marketplace.  I see a lot of examples of extensions that are only TS/JS, but my extension requires python to run too.

Currently installing via VSCode requires a couple extra steps if I remember correctly.  I would like to be a first class extension, and to be able to test the installed packaged via CI... All of this generally works for regular vscode extensions.

Are there any simple examples of this?

**Comment by olguzzar** [comment]
@rebornix may be able to help

**Comment by burkeholland** [comment]
Have you looked at the Python extension source? https://github.com/microsoft/vscode-python

**Reply by olguzzar** [reply]
Hi @paddymul, did you take a look at the Python extension repo? Did it help your scenario?

**Comment by paddymul** [comment]
Sorry, I owe a couple of responses.

I took another look.  It looks the relevant VSCode extension for me is https://github.com/microsoft/vscode-jupyter I hadn't found that extension before.

In the code there, I see some of the messages posted in modals when I install into a vscode installation.  I will file a similar bug with them.

**Comment by paddymul** [comment]
After reading my initial bug report I could have been clearer:  Here is some more context.

 Buckaroo is built as an [IPYWidget](https://github.com/jupyter-widgets/ipywidgets) based extension of the Jupyter interactive computing environment.  Buckaroo works in regular jupyter notebooks, and I have seen it working in VSCode Jupyter notebooks.

VSCode Jupyter notebooks are very similar to Jupyter notebooks, but slightly different in a couple of ways.  One of those ways is how third party IPYWidget extensions interact with the Jupyter/VS-Code-Jupyter environment.  When a user accesses a third party IPYwidgets based extension, a couple of warnings pop up, and the experience isn't smooth.

I would like my project - Buckaroo, to be installable inside of VSCode Jupyter notebooks via the VSCode marketplace.  That was my first question.

My second question is how can I unit test this installablity as a test, that I can run in CI.

Until I looked today, I hadn't been able to find the source code for the vscode-jupyter extension and I figured it was closed source.

**Comment by paddymul** [comment]
Here are the two bugs I filed against vscode-jupyter
https://github.com/microsoft/vscode-jupyter/issues/16438
https://github.com/microsoft/vscode-jupyter/issues/16437
